content/tutorial/04-advanced-sveltekit/06-environment-variables/01-private/README.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Private environment variables
+---
+
+Environment variables — like API keys and database credentials — can be added to a `.env` file, and they will be made available to your application.
+
+> You can also use `.env.local` or `.env.[mode]` files — see the [Vite documentation](https://vitejs.dev/guide/env-and-mode.html#env-files) for more information). Make sure you add any files containing sensitive information to your `.gitignore` file!
+
+In this exercise, we want to allow the user to enter the website if they know the correct passphrase, using an environment variable.
+
+First, in `.env`, add a new environment variable:
+
+```env
+/// file: .env
+PASSPHRASE=+++"open sesame"+++
+```
+
+Open `src/routes/+page.server.js`. Import `PASSPHRASE` from `$env/static/private` and use it inside the [form action](/tutorial/the-form-element):
+
+```js
+/// file: src/routes/+page.server.js
+import { redirect, fail } from '@sveltejs/kit';
++++import { PASSPHRASE } from '$env/static/private';+++
+
+export function load({ cookies }) {
+	if (cookies.get('allowed')) {
```

Comment:
I think we need an example where the environment variable is actually used for something though, otherwise I'm left with the same 'what am I actually doing here?' confusion as the `[page=integer]` example. I couldn't think of any examples (like database connection strings or external API keys) that were realistic to implement in the tutorial. 

It's a good thing if later exercises reinforce things learned in earlier ones — layering the material is the best way for people to develop familiarity (or remind them to revisit past exercises if they find themselves confused by what they're looking at). It reminds people that the `cookies` API is a thing, and it's a reminder of how to use form actions. Take the point about persistence — I added a `maxAge` of `60` for that reason, but we could either a) reduce that further or b) add a `logout` action on the `/welcome` page, which makes the example more complete in any case

Reply:
I don't mean to remove that from the other load function where it errors if you're not "authenticated", only from the "enter the passphrase" page